cmake/api_macros.cmake

```
@@ -206,7 +210,7 @@ macro(OPENDDS_TARGET_SOURCES target)
     message(FATAL_ERROR "OpenDDS does not support argument items in CMAKE_CXX_COMPILER.")
   endif()
 
-  if(OPENDDS_SUPPRESS_ANYS)
+  if("${_suppress_typecodes}" MATCHES ON OR (NOT "${_suppress_typecodes}" MATCHES OFF AND OPENDDS_SUPPRESS_ANYS MATCHES ON))
```

I suggest having

```cmake
set(_suppress_typecodes ${OPENDDS_SUPPRESS_ANYS})
```
before `OPENDDS_GET_SOURCES_AND_OPTIONS` and then here:

```suggestion
  if(_suppress_typecodes)
```